core/main/src/firebolt/handlers/privacy_rpc.rs

```diff
@@ -549,9 +553,22 @@ impl PrivacyImpl {
                     "PrivacySettingsStorageType::Cloud: Not Available",
                 )))
             }
-            // PrivacySettingsStorageType::Sync => Err(jsonrpsee::core::Error::Custom(String::from(
-            //     "PrivacySettingsStorageType::Sync: Unimplemented",
-            // ))),
+            PrivacySettingsStorageType::Sync => {
+                let dist_session = platform_state.session_state.get_account_session().unwrap();
+                let request = PrivacyCloudRequest::SetProperty(SetPropertyParams {
+                    setting: property.as_privacy_setting().unwrap(),
+                    value,
+                    dist_session,
+                });
+                let result = platform_state.get_client().send_extn_request(request).await;
```

Comment:
If I understand things correctly, the cloud updating part is the same both in sync and cloud. Only we need an additional step in case of sync. So can we refactor similar to the below snippet, such that we don't have to write again the cloud updation part? This way we have only one place to update in case of changes in the way we store.

```
match storage_type{
    PrivacySettingsStorageType::Cloud | PrivacySettingsStorageType::Sync => {
        let mut result = cloud_updation_portion();
        if storage_type == PrivacySettingsStorageType::Sync && result.is_ok(){
            result = local_updation_portion();
        }
        result.map_err(|_|{
            Err(jsonrpsee::core::Error::Custom(String::from(
                    "PrivacySettingsStorageType::Sync: Not Available",
            )))
        })
     }
}
```

Reply:
yeah, this is a cleaner way to add additional step in the case of sync.